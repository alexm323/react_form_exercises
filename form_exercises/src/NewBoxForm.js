import React, { useState } from 'react';

const NewBoxForm = ({ addBox }) => {
    const INITIAL_STATE = {
        color: '',
        height: '',
        width: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({ ...formData });
        setFormData({ height: "", width: "", backgroundColor: "" });

    }

    return (
        <form onSubmit={handleSubmit}>

            <label forHtml='color'>Color</label>
            <input id='color' aria-label="Color" name='color' type="text" value={formData.backgroundColor} onChange={handleChange} />


            <label forHtml='height'>Height</label>
            <input id='height' aria-label="Height" name='height' type='text' value={formData.height} onChange={handleChange} />



            <label forHtml='width'>Width</label>
            <input id='width' aria-label="Width" name='width' type='text' value={formData.width} onChange={handleChange} />

            <button>Add Color Box</button>
        </form >
    )
}
export default NewBoxForm;
