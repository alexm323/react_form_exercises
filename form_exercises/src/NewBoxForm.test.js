import React from 'react';
import { render } from '@testing-library/react';
import NewBoxForm from "./NewBoxForm";

// smokebox test 

it("renders without crashing", () => {
    render(<NewBoxForm />);
});
it("matches snapshot", () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});