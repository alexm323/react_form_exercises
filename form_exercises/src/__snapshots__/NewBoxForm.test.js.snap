// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      forhtml="color"
    >
      Color
    </label>
    <input
      aria-label="Color"
      id="color"
      name="color"
      type="text"
      value=""
    />
    <label
      forhtml="height"
    >
      Height
    </label>
    <input
      aria-label="Height"
      id="height"
      name="height"
      type="text"
      value=""
    />
    <label
      forhtml="width"
    >
      Width
    </label>
    <input
      aria-label="Width"
      id="width"
      name="width"
      type="text"
      value=""
    />
    <button>
      Add Color Box
    </button>
  </form>
</DocumentFragment>
`;
